ompi/mca/coll/han/coll_han_allgather.c

```diff
@@ -67,19 +67,28 @@ mca_coll_han_allgather_intra(const void *sbuf, int scount,
 {
     /* Create the subcommunicators */
     mca_coll_han_module_t *han_module = (mca_coll_han_module_t *) module;
-    mca_coll_han_comm_create_new(comm, han_module);
+    if( OMPI_SUCCESS != mca_coll_han_comm_create_new(comm, han_module) ) {
+        OPAL_OUTPUT_VERBOSE((30, mca_coll_han_component.han_output,
+                             "han cannot handle allgather within this communicator. Fall back on another component\n"));
+        /* Put back the fallback collective support and call it once. All
+         * future calls will then be automatically redirected.
+         */
+        comm->c_coll->coll_allgather = han_module->fallback.allgather.allgather;
+        comm->c_coll->coll_allgather_module = han_module->fallback.allgather.module;
+        return comm->c_coll->coll_allgather(sbuf, scount, sdtype, rbuf, rcount, rdtype,
+                                            comm, comm->c_coll->coll_allgather_module);
+    }
     ompi_communicator_t *low_comm = han_module->sub_comm[INTRA_NODE];
     ompi_communicator_t *up_comm = han_module->sub_comm[INTER_NODE];
     int low_rank = ompi_comm_rank(low_comm);
     int w_rank = ompi_comm_rank(comm);
 
     /* Init topo */
     int *topo = mca_coll_han_topo_init(comm, han_module, 2);
-
     /* unbalanced case needs algo adaptation */
-    if (han_module->are_ppn_imbalanced){
+    if (han_module->are_ppn_imbalanced) {
```

Comment:
I believe this check for imbalanced ppn is not necessary for `allgather`

Reply:
After validation this check is necessary. We will keep it for now.